andela/knights-bn-backend
# #170947666 make one travel request has multiple stops

#### What does this PR do?
- Add functionality for letting requester to make one travel request that has multiple destinations embedded.
#### Description of Task to be completed?
- As requester I will be able to request to travel in more that one destination at once without creating separate request for each destination.
#### How should this be manually tested?
- open your terminal and run this command `git clone https://github.com/andela/knightsbn    backend.git` to clone repository
- run `cd knights-bn-backend` to get inside the project
- run `git fetch && git checkout ft-make-multi-city-request-170947666` 
- run `npm install` to install dependencies
- check if you have PostgresSQL database if you don't have it install it from here [PostgreSQL](https://www.2ndquadrant.com/en/blog/pginstaller-install-postgresql/)
- create database
- create `.env` file in your project root directory
- follow `env.example` file to create environment variables in `.env` file and 
   assign them with all your database configurations
- run `npm run createTables` to create tables in your new database you have created.
- run `npm run test` to run all the tests
- run `npm run dev-start` to start the server.
- open your postman or any other API tester you have
-  to create your request you must be logged in or signed up , see how to do [log in](https://github.com/andela/knights-bn-backend/pull/23) or [sign up](https://github.com/andela/knights-bn-backend/pull/17)
- to test multi-city request send POST request to this link `http://localhost:4000/api/v1/trips/request/multicity` with the data object defined like the one below in your postman body.
```
{
    "origin": "New york",
    "destination": "East Africa",
    "departureDate": "2020-04-28",
    "returnDate": "2022-09-30",
   "accommodation": "XYZ campus",
    "type": "multi_way",
    "reason": "Reaseach",
    "passportNumber": "12345677",
    "cities": [
      {
        "name": "kigali",
        "from": "2020-05-02",
        "to": "2020-08-31"
      },
       {
        "name": "Nairob",
        "from": "2020-06-02",
        "to": "2020-09-08"
      }
    ]
  }
```
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947666](https://www.pivotaltracker.com/story/show/170947666)
#### Screenshots

<img width="896" alt="Screen Shot 2020-03-03 at 09 02 56" src="https://user-images.githubusercontent.com/36619897/75750883-d95ec680-5d2d-11ea-8ad6-455cfd532bb5.png">
